src/renderer/dx/DxRenderer.hpp

```diff
@@ -40,98 +35,61 @@ namespace Microsoft::Console::Render
+        [[nodiscard]] HRESULT GetProposedFont(const FontInfoDesired& FontInfoDesired, _Out_ FontInfo& FontInfo, int iDpi) noexcept override;
         [[nodiscard]] HRESULT GetDirtyArea(gsl::span<const til::rectangle>& area) noexcept override;
-
-        [[nodiscard]] HRESULT GetFontSize(_Out_ COORD* const pFontSize) noexcept override;
-        [[nodiscard]] HRESULT IsGlyphWideByFont(const std::wstring_view glyph, _Out_ bool* const pResult) noexcept override;
-
-        [[nodiscard]] ::Microsoft::Console::Types::Viewport GetViewportInCharacters(const ::Microsoft::Console::Types::Viewport& viewInPixels) noexcept;
-        [[nodiscard]] ::Microsoft::Console::Types::Viewport GetViewportInPixels(const ::Microsoft::Console::Types::Viewport& viewInCharacters) noexcept;
-
-        float GetScaling() const noexcept;
-
-        void SetSelectionBackground(const COLORREF color, const float alpha = 0.5f) noexcept;
-        void SetAntialiasingMode(const D2D1_TEXT_ANTIALIAS_MODE antialiasingMode) noexcept;
-        void SetDefaultTextBackgroundOpacity(const float opacity) noexcept;
-        void SetIntenseIsBold(const bool opacity) noexcept;
-
-        void UpdateHyperlinkHoveredId(const uint16_t hoveredId) noexcept;
+        [[nodiscard]] HRESULT GetFontSize(_Out_ COORD* pFontSize) noexcept override;
+        [[nodiscard]] HRESULT IsGlyphWideByFont(std::wstring_view glyph, _Out_ bool* pResult) noexcept override;
+
+        // DxRenderer - getter
+        HRESULT Enable() noexcept override;
+        [[nodiscard]] bool GetRetroTerminalEffect() const noexcept override;
+        [[nodiscard]] float GetScaling() const noexcept override;
+        [[nodiscard]] HANDLE GetSwapChainHandle() override;
+        [[nodiscard]] Types::Viewport GetViewportInCharacters(const Types::Viewport& viewInPixels) const noexcept override;
+        [[nodiscard]] Types::Viewport GetViewportInPixels(const Types::Viewport& viewInCharacters) const noexcept override;
+        // DxRenderer - setter
+        void SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE antialiasingMode) noexcept override;
+        void SetCallback(std::function<void()> pfn) noexcept override;
+        void SetDefaultTextBackgroundOpacity(float opacity) noexcept override;
+        void SetForceFullRepaintRendering(bool enable) noexcept override;
+        [[nodiscard]] HRESULT SetHwnd(HWND hwnd) noexcept override;
+        void SetPixelShaderPath(std::wstring_view value) noexcept override;
+        void SetRetroTerminalEffect(bool enable) noexcept override;
+        void SetSelectionBackground(COLORREF color, float alpha = 0.5f) noexcept override;
+        void SetSoftwareRendering(bool enable) noexcept override;
+        void SetIntenseIsBold(bool enable) noexcept override;
+        void SetWarningCallback(std::function<void(HRESULT)> pfn) noexcept override;
+        [[nodiscard]] HRESULT SetWindowSize(SIZE pixels) noexcept override;

```

Comment:
Type qualifiers (`const` and `volatile`) don't hold any meaning for declarations of parameters and return types.

`void foo(int)`, `void foo(int i)`, and `void foo(const int i)`, and `void foo(volatile int i)` are all identical function declarations and this is valid code that will compile:
```cpp
const int foo(const int parameter);

int foo(int parameter)
{
    parameter *= 2;
    return parameter;
}
```

If you pass a parameter by-value it doesn't really matter to the caller whether the receiver modifies the copy, right? This is why `const` holds no meaning in declarations and is thus code you don't need.

The reason I remove those _if_ I touch the offending code (in this case because I copy-pasted the new `IRenderEngine` interface into DxEngine.h -- I didn't actually intend to touch DxEngine itself) is because all compilers except for MSVC warn about this and it's just a matter of time until when we get similar warnings in this project (gcc for instance warns about this in pedantic mode as `-Wignored-qualifiers` and `clang-tidy` warns about it by default).